code/game/objects/items/weapons/pet_carrier.dm

```diff
@@ -0,0 +1,262 @@
+		to_chat(user, SPAN_NOTICE("You start pushing out of [src]... (This will take about 20 seconds.)"))
+		if(!do_after(user, 200, target = user) || open || !locked || !(user in occupants))
+			return
+		loc.visible_message(SPAN_WARNING("[user] shoves out of [src]!"))
+		to_chat(user, SPAN_NOTICE("You shove open [src]'s door against the lock's resistance and fall out!"))
+		locked = FALSE
+		open = TRUE
+		//update_appearance()
+		update_icon()
+		remove_occupant(user)
+
+/obj/item/weapon/pet_carrier/update_icon()
+	if(open)
+		icon_state = initial(icon_state)
+		return ..()
+	icon_state = "[base_icon]_[!occupants.len ? "closed" : "occupied"]"
+	cut_overlays()
+	add_overlay("[base_icon]_[locked ? "" : "un"]locked")
+	return ..()
+/*
+/obj/item/pet_carrier/update_overlays()
+	. = ..()
+	if(!open)
+		. += "[icon_state]_[locked ? "" : "un"]locked"
+// This wont fuckin work, and I don't care to fix it. I'll just make a verb do the same thing.
+/obj/item/pet_carrier/MouseDrop(atom/over_atom)
+	. = ..()
+	var/turf/T = get_turf(over_atom.loc)
+	if(T.check_density(ignore_mobs = TRUE) && usr.CanUseTopic(src, TRUE) && usr.Adjacent(over_atom) && open && occupants.len)
+		usr.visible_message(span_notice("[usr] unloads [src]."), \
+		span_notice("You unload [src] onto [over_atom]."))
+		for(var/V in occupants)
+			remove_occupant(V, T)
+*/
+/obj/item/weapon/pet_carrier/proc/drop_pet()
+	set name = "Empty Carrier"
+	set category = "Object"
+	set src in view(1)
+
+	for(var/V in occupants)
+		remove_occupant(V)
+
+/obj/item/weapon/pet_carrier/proc/load_occupant(mob/living/user, mob/living/target)
+	if(pet_carrier_full(src))
+		to_chat(user, SPAN_WARNING("[src] is already carrying too much!"))
+		return
+	user.visible_message(SPAN_NOTICE("[user] starts loading [target] into [src]."), \
+	SPAN_NOTICE("You start loading [target] into [src]..."))
+	to_chat(target, SPAN_DANGER("[user] starts loading you into \the [name]!"))
+	if(!do_mob(user, target, 30))
+		return
+	if(target in occupants)
+		return
+	if(pet_carrier_full(src)) //Run the checks again, just in case
+		to_chat(user, SPAN_WARNING("[src] is already carrying too much!"))
+		return
+	user.visible_message(SPAN_NOTICE("[user] loads [target] into [src]!"), \
+	SPAN_NOTICE("You load [target] into [src]."))
+	to_chat(target, SPAN_DANGER("[user] loads you into \the [name]!"))
+	add_occupant(target)
+
+/obj/item/weapon/pet_carrier/proc/add_occupant(mob/living/occupant)
+	if(occupant in occupants || !istype(occupant))
+		return
+	occupant.forceMove(src)
+	occupants += occupant
+	occupant_weight += occupant.mob_size
+
+/obj/item/weapon/pet_carrier/proc/remove_occupant(mob/living/occupant, turf/new_turf)
+	if(!(occupant in occupants) || !istype(occupant))
+		return
+	occupant.forceMove(new_turf ? new_turf : drop_location())
+	if(occupant.buckled && istype(occupant.buckled, /obj/effect/energy_net))
+		occupant.buckled.dropInto(src)
+	occupants -= occupant
+	occupant_weight -= occupant.mob_size
+	occupant.dir = SOUTH
```

Comment:
use setDir so relevant COMSIG's can fire (assuming upstream implemented that)

Reply:
setDir doesn't exist in this codebase, unless it's called something goofy and I can't find it.

Reply:
Necro post, but setDir is set_dir on here